plugins/module_utils/network/sonic/config/ipv6_router_advertisement/ipv6_router_advertisement.py

```diff
@@ -0,0 +1,587 @@
+                        rdnss_payload = self.get_ipv6_rtadv_rdnss_payload(value)
+                        if rdnss_payload:
+                            rtadv_payload[OPTION_TO_PAYLOAD_MAP[option]] = rdnss_payload
+                    elif option == 'router_preference':
+                        rtadv_config_payload[OPTION_TO_PAYLOAD_MAP[option]] = 'openconfig-interfaces-ext:' + value.upper()
+                    else:
+                        rtadv_config_payload[OPTION_TO_PAYLOAD_MAP[option]] = value
+
+            if rtadv_config_payload:
+                rtadv_payload['config'] = rtadv_config_payload
+            if rtadv_payload:
+                requests.append({'path': rtadv_path, 'method': PATCH, 'data': {'openconfig-if-ip:router-advertisement': rtadv_payload}})
+
+        return requests
+
+    @staticmethod
+    def get_ipv6_rtadv_dnssl_payload(dnssl_list):
+        payload = []
+        for dnssl in dnssl_list:
+            config_payload = {'dnssl-name': dnssl['dnssl_name']}
+            if 'valid_lifetime' in dnssl:
+                config_payload['valid-lifetime'] = dnssl['valid_lifetime']
+            payload.append({'dnssl-name': dnssl['dnssl_name'], 'config': config_payload})
+
+        if payload:
+            return {'dns-search-name': payload}
+
+        return None
+
+    @staticmethod
+    def get_ipv6_rtadv_ra_prefixes_payload(prefix_list):
+        payload = []
+        for prefix in prefix_list:
+            config_payload = {'prefix': prefix['prefix']}
+            for option in ('no_autoconfig', 'off_link', 'preferred_lifetime', 'router_address', 'valid_lifetime'):
+                if option in prefix:
+                    config_payload[option.replace('_', '-')] = prefix[option]
+            payload.append({'prefix': prefix['prefix'], 'config': config_payload})
+
+        if payload:
+            return {'ra-prefix': payload}
+
+        return None
+
+    @staticmethod
+    def get_ipv6_rtadv_rdnss_payload(rdnss_list):
+        payload = []
+        for rdnss in rdnss_list:
+            config_payload = {'address': rdnss['address']}
+            if 'valid_lifetime' in rdnss:
+                config_payload['valid-lifetime'] = rdnss['valid_lifetime']
+            payload.append({'address': rdnss['address'], 'config': config_payload})
+
+        if payload:
+            return {'rdnss-address': payload}
+
+        return None
+
+    def get_rtadv_path(self, intf_name):
+        if intf_name.startswith('Vlan'):
+            return self.vlan_rtadv_path.format(vlan_name=intf_name)
+        else:
+            sub_intf = 0
+            if '.' in intf_name:
+                intf_name, sub_intf = intf_name.split('.')
+            return self.non_vlan_rtadv_path.format(intf_name=intf_name, sub_intf=sub_intf)
+
+    def get_diff(self, want, have):
+        updated_want = []
+        have_dict = {item['name']: item for item in have}
+        for conf in want:
+            have_conf = have_dict.get(conf['name'], {})
```

Comment:
Two lines following this one appear to have been accidentally deleted when changing this instruction to use the comprehended "have_dict". Although the default value specified for the "get" above will prevent an exception, it would be more efficient to keep the conditional "continue" here for the case when the "get" result is empty.

Please restore those two instructions as suggested here:

```suggestion
            have_conf = have_dict.get(conf['name'], {})
            if not have_conf:
                continue
                 
```